src/main/java/com/commercetools/sync/services/impl/BaseService.java

```diff
@@ -1,32 +1,50 @@
 package com.commercetools.sync.services.impl;
 
 import com.commercetools.sync.commons.BaseSyncOptions;
+import com.commercetools.sync.commons.utils.CtpQueryUtils;
+import io.sphere.sdk.client.SphereClient;
 import io.sphere.sdk.commands.DraftBasedCreateCommand;
 import io.sphere.sdk.commands.UpdateAction;
 import io.sphere.sdk.commands.UpdateCommand;
 import io.sphere.sdk.models.Resource;
+import io.sphere.sdk.models.WithKey;
+import io.sphere.sdk.queries.MetaModelQueryDsl;
+import org.apache.commons.lang3.StringUtils;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.BiFunction;
+import java.util.function.Consumer;
 import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
 
 import static com.commercetools.sync.commons.utils.SyncUtils.batchElements;
 import static java.lang.String.format;
 import static org.apache.commons.lang3.StringUtils.isBlank;
 
 /**
  * @param <T> Resource Draft (e.g. {@link io.sphere.sdk.products.ProductDraft},
- *  {@link io.sphere.sdk.categories.CategoryDraft}, etc..
+ *            {@link io.sphere.sdk.categories.CategoryDraft}, etc..
  * @param <U> Resource (e.g. {@link io.sphere.sdk.products.Product}, {@link io.sphere.sdk.categories.Category}, etc..
  * @param <S> Subclass of {@link BaseSyncOptions}
+ * @param <Q> Query (e.g. {@link io.sphere.sdk.products.queries.ProductQuery},
+ *            {@link io.sphere.sdk.categories.queries.CategoryQuery}, etc..
```

Comment:
why not closing the brackets? 😄 
```suggestion
 *            {@link io.sphere.sdk.categories.queries.CategoryQuery}, etc.. )
```

Reply:
@heshamMassoud  Please update others too